test/e2e/autoscaling.go

```diff
@@ -0,0 +1,161 @@
+		// Ensure cluster size is equal to 1.
+		expectNoError(waitForClusterSize(f.Client, 1))
+	})
+
+	AfterEach(func() {
+		cleanUpAutoscaler()
+	})
+
+	It("[Skipped] [Autoscaling] should scale cluster size based on cpu utilization", func() {
+		setUpAutoscaler("cpu/node_utilization", 0.8, 1, 10)
+
+		ConsumeCpu(f, "cpu-utilization", 1)
+		expectNoError(waitForClusterSize(f.Client, 2))
+
+		StopConsuming(f, "cpu-utilization")
+		expectNoError(waitForClusterSize(f.Client, 1))
+	})
+
+	It("[Skipped] [Autoscaling] should scale cluster size based on cpu reservation", func() {
+		setUpAutoscaler("cpu/node_reservation", 0.7, 1, 10)
+
+		ReserveCpu(f, "cpu-reservation", 800)
+		expectNoError(waitForClusterSize(f.Client, 2))
+
+		StopConsuming(f, "cpu-reservation")
+		expectNoError(waitForClusterSize(f.Client, 1))
+	})
+
+	It("[Skipped] [Autoscaling] should scale cluster size based on memory utilization", func() {
+		setUpAutoscaler("memory/node_utilization", 0.5, 1, 10)
+
+		ConsumeMemory(f, "memory-utilization", 2)
+		expectNoError(waitForClusterSize(f.Client, 2))
+
+		StopConsuming(f, "memory-utilization")
+		expectNoError(waitForClusterSize(f.Client, 1))
+	})
+
+	It("[Skipped] [Autoscaling] should scale cluster size based on memory reservation", func() {
+		setUpAutoscaler("memory/node_reservation", 0.5, 1, 10)
+
+		ReserveMemory(f, "memory-reservation", 2000)
+		expectNoError(waitForClusterSize(f.Client, 2))
+
+		StopConsuming(f, "memory-reservation")
+		expectNoError(waitForClusterSize(f.Client, 1))
+	})
+})
+
+func setUpAutoscaler(metric string, target float64, min, max int64) {
+	By("Setting up autoscaler to scale based on " + metric)
+	_, err := exec.Command("gcloud", "preview", "autoscaler",
+		"--zone="+testContext.CloudConfig.Zone,
+		"create", "e2e-test-autoscaler",
+		"--project="+testContext.CloudConfig.ProjectID,
+		"--target="+testContext.CloudConfig.NodeInstanceGroup,
+		"--custom-metric=custom.cloudmonitoring.googleapis.com/kubernetes.io/"+metric,
+		fmt.Sprintf("--target-custom-metric-utilization=%v", target),
+		"--custom-metric-utilization-target-type=GAUGE",
+		fmt.Sprintf("--min-num-replicas=%v", min),
+		fmt.Sprintf("--max-num-replicas=%v", max),
+	).CombinedOutput()
+	expectNoError(err)
+}
+
+func cleanUpAutoscaler() {
+	By("Removing autoscaler")
+	_, err := exec.Command("gcloud", "preview", "autoscaler", "--zone="+testContext.CloudConfig.Zone, "delete", "e2e-test-autoscaler").CombinedOutput()
+	expectNoError(err)
+}
+
+func ConsumeCpu(f *Framework, id string, cores int) {
+	By(fmt.Sprintf("Running RC which consumes %v cores", cores))
+	config := &RCConfig{
+		Client:    f.Client,
+		Name:      id,
+		Namespace: f.Namespace.Name,
+		Timeout:   10 * time.Minute,
+		Image:     "jess/stress",
+		Command:   []string{"stress", "-c", "1"},
+		Replicas:  cores,
+	}
+	expectNoError(RunRC(*config))
+}
+
+func ConsumeMemory(f *Framework, id string, gigabytes int) {
+	By(fmt.Sprintf("Running RC which consumes %v GB of memory", gigabytes))
```

Comment:
There is no possibility to consume fractional part of the core on the other hand it is not possible to schedule pod which limit equals to 1 core so that the API is not consistent. Since it is temporary solution until #11570 is done let's leave it as it is now.

I'll change memory reservation to use gigabytes instead.